core/src/mos6502.cpp

```diff
@@ -609,6 +613,25 @@ Pipeline Mos6502::create_eor_instruction(Opcode opcode) {
     return result;
 }
 
+Pipeline Mos6502::create_rol_instruction(Opcode opcode) {
+    const MemoryAccess memory_access = get_memory_access(opcode.family);
+    Pipeline result;
+    result.append(create_addressing_steps(opcode.address_mode, memory_access));
+
+    result.push([=]() {
+        const uint8_t operand = mmu_->read_byte(effective_address_);
+        const uint8_t carry = registers_->p & C_FLAG ? 0x01 : 0x00;
+        const uint16_t temp_result = (operand << 1) | carry;
+        registers_->a = static_cast<uint8_t>(temp_result);
```

Comment:
Looks like only the Accumulator addressing mode for ROL should modify a. The rest should read from memory, and write back to the same memory address.

Reply:
Ok, need to do my homework a bit more...